lib/takeaway.rb

```diff
@@ -0,0 +1,69 @@
+require 'twilio-ruby'
+require 'dotenv'
+require_relative 'menu'
+require_relative 'order'
+
+class Takeaway
+  DELIVERY_TIME = 1
+
+  def initialize(phone, menu = Menu.new, order = Order.new)
+    @phone = phone
+    @menu = menu
+    @order = order
+  end
+
+  def read_menu
+    menu.display_menu
+  end
+
+  def select_dishes(dishes)
+    dishes.each do |item, quantity|
+      order.add(item, quantity) if menu.has_dish?(item)
+    end
+    order.basket
+  end
+
+  def checkout
+    total = order.order_total
+    "You\'ve ordered #{total} items."
+  end
+
+  def confirm_order(number)
+    raise 'Total number of dishes does not match. Please check again.' unless total_correct?(number)
```

Comment:
Metrics/LineLength: Line is too long. [100/80] (https://github.com/bbatsov/ruby-style-guide#80-character-limits)<br>Style/SignalException: Use `fail` instead of `raise` to signal exceptions. (https://github.com/bbatsov/ruby-style-guide#fail-method)